misc/error_injector/malloc/malloc_ei.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+ */
+
+#include <malloc_ei.h>
+
+extern "C" {
+void *__real_malloc(size_t);
```

Comment:
Just to check. the symbol `__real_malloc` which pont to `malloc` due to ld `--wrap`?
And the any usage of `malloc` will result in symbol `__warp_malloc` due to ld `--wrap`?

Reply:
Yes, that's the way it works.